graphs/possible_bipartition.py

```diff
@@ -1,12 +1,46 @@
-# Can be used for BFS
-from collections import deque 
+from collections import defaultdict  
 
 def possible_bipartition(dislikes):
     """ Will return True or False if the given graph
         can be bipartitioned without neighboring nodes put
         into the same partition.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(V+E)
+        Space Complexity: O(V+E)
     """
-    pass
+    # create an adjacency list to represent the graph
+    graph = defaultdict(list)
+    for dislike in dislikes:
+        # if there are no enemies, no need to add to graph
```

Comment:
This is a great insight! Just note that if a dog has no dislikes `len(dislike)` should be equal to 0, not 1 or less. See more explanation below ⬇️